build/ClientAnnotator.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+use Aws\Api\ApiProvider;
+
+class ClientAnnotator
+{
+    use PhpFileLinterTrait;
+
+    /** @var ReflectionClass */
+    private $reflection;
+
+    public function __construct($clientClassName)
+    {
+        $this->reflection = new ReflectionClass($clientClassName);
+    }
+
+    public function updateApiMethodAnnotations()
+    {
+        // copy the code into memory
+        $backup = file($this->reflection->getFileName());
+        $classLineOffset = $this->reflection->getStartLine() - 1;
+        list($preamble, $class) = [
+            implode('', array_slice($backup, 0, $classLineOffset)),
+            implode('', array_slice($backup, $classLineOffset)),
+        ];
+
+        // strip out the existing docBlock if it exists
+        if ($this->reflection->getDocComment()) {
+            $preamble = str_replace(
+                $this->reflection->getDocComment() . PHP_EOL,
+                '',
+                $preamble
+            );
+        }
+
+        $preamble .= $this->buildDocBlock();
+
+        $this->writeClassFile(implode(PHP_EOL, [$preamble, $class]));
+        if (!$this->lintFile($this->reflection->getFileName())) {
+            $this->writeClassFile($backup);
+        }
+    }
+
+    private function writeClassFile($contents)
+    {
+        return (bool) file_put_contents(
+            $this->reflection->getFileName(),
+            $contents,
+            LOCK_EX
+        );
+    }
+
+    private function buildDocBlock()
+    {
+        $docBlock = $this->reflection->getDocComment()
+            ?: $this->getDefaultDocComment();
+
+        $docLines = explode(PHP_EOL, $docBlock);
+
+        // remove all API @method annotations
+        $docLines = array_filter($docLines, function ($line) {
+            return !preg_match('/^\* @method \\\\Aws\\\\Result /', trim($line));
+        });
+
+        // hold on to the closing line
+        $lastLine = array_pop($docLines);
+
+        // add a padding line if needed
+        if (' *' !== end($docLines)) {
+            $docLines []= ' *';
+        }
+
+        // append API @method annotations
+        $docLines = array_merge($docLines, array_map(function (array $op) {
+            return " * @method {$op['return']} {$op['signature']} {$op['doc']}";
```

Comment:
What's the motivation for adding the description of the method in the method annotation? It adds a lot of extra weight to each client? I'm not even sure where this text would show up in the editor. I'd prefer the doc be used for just listing which versions of an API the method is compatible with.